0012-unify-pulse-commands-and-instructions.md

```diff
@@ -0,0 +1,334 @@
+We will remove the current pulse `Instruction` and replace it with an implementation that is a class with an `operands` attribute that contains all operands that the instruction acts on, including both pulses and constants such as floats.
+
+The `Command` will be deprecated. A `Pulse` type will be defined and all pulses will inherit from this.
+
+`Command`s and `Instruction`s will be combined into a new `Instruction` as shown below:
+
+| `Command`      | old `Instruction`         | new `Instruction` | new Operands                                                                                                                 |
+| -------------- | ------------------------- | ----------------- | ---------------------------------------------------------------------------------------------------------------------------- |
+| `PulseCommand` | `PulseInstruction`        | `Play`            | `pulse:Pulse, channel: PulseChannel`                                                                                         |
+| `Delay`        | `DelayInstruction`        | `Delay`           | `duration: int, channel: Channel`                                                                                            |
+| `FrameChange`  | `FrameChangeInstruction`  | `ShiftPhase`      | `phase: float, channel: PulseChannel`                                                                                        |
+| N/A            | N/A                       | `SetPhase`        | `phase: float, channel: PulseChannel`                                                                                        |
+| N/A            | N/A                       | `ShiftFrequency`  | `frequency: float, channel: PulseChannel`                                                                                    |
+| `SetFrequency` | `SetFrequencyInstruction` | `SetFrequency`    | `frequency: float, channel: PulseChannel`                                                                                    |
+| `Acquire`      | `AcquireInstruction`      | `Acquire`         | `duration: int, register: Union[MemorySlot, RegisterSlot], kernel: Optional[Kernel], discriminator: Optional[Discriminator]` |
+| `Snapshot`     | `Snapshot`                | `Snapshot`        | `label: str, snapshot_type: str`                                                                                             |
+
+### Example of pulse programming with changes
+```python
+x90 = gaussian(duration, amp, sigma)
+meas_stim = gaussian_square(duration, amp, sigma, width)
+
+sched = Schedule()
+# we will keep callable shorthand for pulses
+sched += x90(DriveChannel(0))
+sched += Delay(100, DriveChannel(0))
+# or we can generate a Play instruction explicitly
+sched += Play(x90, DriveChannel(0))
+# measurements may be scheduled with delay
+sched += Delay(200, AcquireChannel(0))
+sched += Play(meas_stim, MeasureChannel(0))
+# or shifting in time
+sched += Acquire(100, AcquireChannel(0), MemorySlot(0)) << 200
```

Comment:
Conflicting with the table above. The table says the new Acquire instruction have no AcquireChannel in its operands. Either should be fixed.